### ISSUE TITLE
Compatibility Issue Between eslint-ts-patch and ESLint v8.57.0

### ISSUE THREAD

**Issue by cwandev** [antfu/eslint-ts-patch#5]
### Describe the bug

When following the installation instructions step by step, the default installation resulted in the latest version of ESLint, v8.57.0, being installed. However, it failed to function properly alongside the eslint-ts-patch. Upon installing ESLint v8.56.0 instead, the patch started working as expected. I am unsure of the reason behind this discrepancy, but reverting to ESLint v8.56.0 did resolve the issue. 

Could I have missed any crucial instructions or made any mistakes during the installation process? Any insights or assistance would be greatly appreciated. Thank you for your attention to this matter.

### Reproduction

*

### System Info

```Shell
System:
  OS: Windows 11 10.0.22621
  CPU: (12) x64 12th Gen Intel(R) Core(TM) i5-12400
  Memory: 16.20 GB / 31.75 GB
Binaries:
  Node: 20.8.1 - C:\Program Files\nodejs\node.EXE
  npm: 10.1.0 - C:\Program Files\nodejs\npm.CMD
  pnpm: 8.15.4 - C:\Program Files\nodejs\pnpm.CMD
Browsers:
  Edge: Chromium (122.0.2365.59)
  Internet Explorer: 11.0.22621.1
```

### Used Package Manager

pnpm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/antfu/.github/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://github.com/antfu/contribute).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A, please open a GitHub Discussion instead.
- [x] The provided reproduction is a [minimal reproducible](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by alexzhang1030** [comment]
The same issue is faced, and the `vscode-eslint` plugin will throw an error like `Cannot find config file` when using `eslint.config.ts`, but `eslint.config.js` is well.

**Comment by mancuoj** [comment]
same issue 

```log
Oops! Something went wrong! :(

ESLint: 8.57.0

ESLint couldn't find a configuration file. To set up a configuration file for this project, please run:

    npm init @eslint/config

ESLint looked for configuration files in /home/mancuoj/Projects/react-tmpl and its ancestors. If it found none, it then looked in your home directory.

If you think you already have a configuration file or if you need more help, please stop by the ESLint Discord server: https://eslint.org/chat
```

**Comment by yunsii** [comment]
Fixed in v8.57.0-0 https://github.com/antfu/eslint-ts-patch/commit/c50680205ed393d8dceea664a68e8209e9bd8a37
